include/boost/geometry/algorithms/line_interpolate.hpp

```diff
@@ -0,0 +1,410 @@
+                                                               strategy);
+    }
+};
+
+template <BOOST_VARIANT_ENUM_PARAMS(typename T)>
+struct line_interpolate<boost::variant<BOOST_VARIANT_ENUM_PARAMS(T)> >
+{
+    template <typename Pointlike, typename Strategy>
+    struct visitor: boost::static_visitor<void>
+    {
+        Pointlike const& m_pointlike;
+        Strategy const& m_strategy;
+
+        visitor(Pointlike const& pointlike, Strategy const& strategy)
+            : m_pointlike(pointlike)
+            , m_strategy(strategy)
+        {}
+
+        template <typename Geometry, typename Distance>
+        void operator()(Geometry const& geometry, Distance const& max_distance) const
+        {
+            line_interpolate<Geometry>::apply(geometry, max_distance,
+                                                    m_pointlike, m_strategy);
+        }
+    };
+
+    template <typename Distance, typename Pointlike, typename Strategy>
+    static inline void
+    apply(boost::variant<BOOST_VARIANT_ENUM_PARAMS(T)> const& geometry,
+          double const& max_distance,
+          Pointlike & pointlike,
+          Strategy const& strategy)
+    {
+        boost::apply_visitor(
+            visitor<Pointlike, Strategy>(pointlike, strategy),
+            geometry,
+            max_distance
+        );
+    }
+};
+
+} // namespace resolve_variant
+
+/*!
+\brief 	Returns one or more points interpolated along a LineString \brief_strategy
+\ingroup line_interpolate
+\tparam Geometry Any type fulfilling a LineString concept
+\tparam Distance A numerical distance measure
+\tparam Pointlike Any type fulfilling Point or Multipoint concept
+\tparam Strategy A type fulfilling a LineInterpolatePointStrategy concept
+\param geometry Input geometry
+\param max_distance Distance threshold (in units depending on coordinate system)
+representing the spacing between the points
+\param pointlike Output: either a Point (exactly one point will be constructed) or
+a MultiPoint (depending on the max_distance one or more points will be constructed)
+\param strategy line_interpolate strategy to be used for interpolation of
+points
+
+\qbk{[include reference/algorithms/line_interpolate.qbk]}
+
+\qbk{distinguish,with strategy}
+
+\qbk{
+[heading Available Strategies]
+\* [link geometry.reference.strategies.strategy_line_interpolate_cartesian Cartesian]
+\* [link geometry.reference.strategies.strategy_line_interpolate_spherical Spherical]
+\* [link geometry.reference.strategies.strategy_line_interpolate_geographic Geographic]
+
+[heading Example]
+[line_interpolate_strategy]
+[line_interpolate_strategy_output]
+
+[heading See also]
+\* [link geometry.reference.algorithms.densify densify]
+}
+ */
+template
+<
+    typename Geometry,
+    typename Distance,
+    typename Pointlike,
+    typename Strategy
+>
+inline void line_interpolate(Geometry const& geometry,
+                                   Distance const& max_distance,
```

Comment:
Indentation could be improved here.

Reply:
Thanks, I fixed this along with few others